ark/spLDA/processing.py

```diff
@@ -63,24 +52,153 @@ def format_cell_table(cell_table, markers=None, clusters=None, fovs="all_fovs"):
+                * marker: for each marker, count the total number of cells
+                within a ``radius`` *r* from cell *i*
+                having marker expression greater than 0.5.
+                * avg_marker: for each marker, compute the average marker
+                expression of all cells with a ``radius`` *r*
+                from cell *i*.
+                * cluster: for each cluster, count the total number of cells
+                with a ``radius`` *r* from cell *i*
+                belonging to that cell cluster.
+                * count: counts the total number of cells within a ``radius``
+                *r* from cell *i*.
+        radius (int):
+            Size of the radius, in pixels, used to featurize cellular
+            neighborhoods.
+        cell_index (str):
+            Name of the column containing the reference cell indexes.  If not
+            specified, all cells are used.
+        n_processes (int):
+            Number of parallel processes to use.
+        train_split (bool):
+            Create an index for training data in each field of view.
 
+    Returns:
+        dict:
 
+        - A dictionary containing featurized cellular neighborhoods and
+        training splits if applicable.
+    """
+    check_featurize_cell_table_args(cell_table=cell_table,
+                                    feature_by=feature_by, radius=radius,
+                                    cell_index=cell_index)
+    # Define Featurization Function
+    func_type = {
+        "marker": neighborhood_to_marker,
+        "cluster": neighborhood_to_cluster,
+        "avg_marker": neighborhood_to_avg_marker,
+        "count": neighborhood_to_count
+    }
+    if feature_by in ["marker", "avg_marker"]:
+        neighborhood_feature_fn = functools.partial(func_type[feature_by],
+                                                    markers=cell_table[
+                                                        "markers"])
+    else:
+        neighborhood_feature_fn = functools.partial(func_type[feature_by])
+
+    # Featurize FOVs
+    feature_sample = {k: v for (k, v) in cell_table.items() if
+                      k in cell_table["fovs"].tolist()}
+    featurized_fovs = featurize_samples(feature_sample,
+                                        neighborhood_feature_fn,
+                                        radius=radius,
+                                        is_anchor_col=cell_index,
+                                        x_col='x',
+                                        y_col='y',
+                                        n_processes=n_processes,
+                                        include_anchors=True)
+
+    if train_split:
+        all_sample_idxs = featurized_fovs.index.map(lambda x: x[0])
+        _sets = train_test_split(featurized_fovs, test_size=1. - train_frac,
+                                 stratify=all_sample_idxs)
+        train_features_fraction, _ = _sets
```

Comment:
Any reason we're storing the `_sets` variable?  Could we just have:
```
train_features_fraction, _ = train_test_split(...)
```

Reply:
Also pulled from Calico notebook, and yes i can change it to be that way.

Reply:
Is there a scenario where we wouldn't do a train split? We could just set train_frac to 1, right?